### PR TITLE
Stop removing focus outlines by default

### LINKED ISSUE #63: Do not remove focus styles by default
## Is your feature request related to a problem? Please describe.
The default behavior when importing `essentials.css` is to remove focus styles from everything on the page. This makes it way too easy for devs to accidentally remove focus styles from elements that do not have a focus visible fallback. Most will not notice that the focus styles are missing, resulting in a major accessibility violation. 

See also AlaskaAirlines/auro-button#97, where focus styles do not properly apply to a custom element inside a shadow root. Because the focus visible polyfill did not load for the inner component, the focus styles inside the entire component are missing. If the default was to show focus styles and only remove them once focus visible was loaded, this issue would not have an a11y impact.

It is better for a page's design to be wrong (focus styles shown when not using the keyboard) than for a page to be inaccessible to keyboard users. Turning off all focus styles by default is an a11y footgun and should be changed.

It is better for focus styles to accidentally be shown than to accidentally be removed. The former means the visual design is incorrect, the latter means the page is inaccessible to keyboard users. Turning off all focus styles by default is an a11y footgun and should be changed.

## Describe the solution you'd like
We should not remove focus styles by default. We should only selectively remove focus styles when the polyfill is loaded, as suggested in the [focus-visible polyfill README](https://github.com/WICG/focus-visible#2-update-your-css).

```css
/*
  This will hide the focus indicator if the element receives focus via the mouse,
  but it will still show up on keyboard focus.
*/
.js-focus-visible :focus:not(.focus-visible) {
  outline: none;
}
```

We could do something similar with the native `:focus-visible` selector since both [Chrome and Firefox](https://caniuse.com/css-focus-visible) have support.

## Describe alternatives you've considered
Rely on education to solve this problem. However, education will not be 100% effective and missing focus styles will slip through the cracks.  Instead, we should provide an accessible default and educate how to use focus-visible to meet our design goals.

### PR DESCRIPTION
**Pull request by geoffrich**
# Alaska Airlines Pull Request

Fixes #63 

### Focus outline changes
This PR stops removing focus outlines by default (see my rationale in the linked issue). The outline is only removed when the browser supports the native :focus-visible selector (or the polyfill is loaded) and the focus-visible state does not apply. I added a rule that uses native :focus-visible since support has landed in Chrome + Firefox, with the intention to eventually remove the polyfill selectors entirely once focus-visible lands in WebKit.

There were also some -moz-focus prefixes in normalize that I removed because they were interfering with the new focus rules. I tested with the latest Firefox (86) and could not discern any differences or rogue focus outlines with these rules removed. I believe [-moz-focusring](https://developer.mozilla.org/en-US/docs/Web/CSS/:-moz-focusring) has been replaced with :focus-visible. 

There is one potential breakage with this release -- outline is no longer removed by default when focusing an element and focus-visible **does** apply. Previously, this was happening because outline was removed in all focus cases. You will need to remove outline in your focus-visible styles yourself if you were using border or box-shadow for focus. I think this should mainly be a concern for custom elements that consume essentials.css, and we can make the necessary changes when updating the packages in those elements. CSS is generated at build time, so nothing should happen unexpectedly for existing consumers of these elements.

### Other changes
- added a local demo page to see how these rules affected native elements and auro-button.
- fixed external link typo in docs

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: src/_core.scss [hunk 1]
```diff
@@ -67,3 +60,15 @@ $focus: null !default;
   // Only visible when Windows High-Contrast Mode is turned on
   outline: 3px solid transparent;
```
Comment (littleninja) [inline src/_core.scss:61]: Could we merge these two selectors?

```scss
.js-focus-visible :focus:not(.focus-visible),
:focus: not(:focus-visible) {
  outline: 3px solid transparent;
}
```

Comment (geoffrich) [reply src/_core.scss:61]: Unfortunately not — a single invalid selector [invalidates the whole rule](https://developer.mozilla.org/en-US/docs/Web/CSS/Selector_list#selector_list_invalidation). So browsers that don’t support :focus-visible would be broken.

Comment (littleninja) [reply src/_core.scss:61]: Ooo, did not realize that could happen if browsers do not support a selector (if I understand your explanation). Thank you!

**Review by littleninja — APPROVED** [review]

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**